stdlib/lazy.mli

```diff
@@ -57,30 +58,36 @@ type 'a t = 'a CamlinternalLazy.t
 
 exception Undefined
 
-exception RacyLazy
-
 (* val force : 'a t -> 'a  *)
 external force : 'a t -> 'a = "%lazy_force"
 (** [force x] forces the suspension [x] and returns its result.
     If [x] has already been forced, [Lazy.force x] returns the
     same value again without recomputing it.  If it raised an exception,
     the same exception is raised again.
     Raise {!Undefined} if the forcing of [x] tries to force [x] itself
-    recursively.
-    Raise {!RacyLazy} if [x] is concurrently forced by another domain.
+    recursively, or if [x] is concurrently forced by another domain.
 *)
 
+val try_force : 'a t -> 'a option
+(** [try_force x] behaves similarly to [Some (force x)], except that
+    it returns immediately with [None] if [x] is concurrently forced by
```

Comment:
```suggestion
    it returns immediately with [None] if [x] is already being forced concurrently and by
```

This distinction is important since the first `force` does not return `None`, only the subsequent concurrent forcing return `None`.

Reply:
ok